bin/find-never-defined-css-variables/find-never-defined-css-variables.ts

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env tsx
+/* eslint-disable no-console */
+import { readFileSync } from "fs";
+
+import { glob } from "glob";
+
+/**
+ * This script finds css variables that are used but *never* defined in our codebase.
+ * Its goal isn't to find usages of variables in contexts where that variable is not defined,
+ * as that would be a lot harder.
+ * Its goal is to prevent typos and wrong usages created by conflicts and backports.
+ * The most common thing this should catch and prevent is backporting a fix or feature that uses a css variable
+ * that doesn't exist in the release branch.
+ */
+
+const shouldWhiteList = (_variable: string) => {
+  // Use this for whitelisting some variables if we know they're defined by a 3rd party, such as mantine
+  // Note: for be sure to not backport changes to this function that would whitelist variables not on the release branch
+  return false;
+};
+
+interface UsageMap {
+  // variable : files
+  [variable: string]: string[];
+}
+
+const findFiles = (): string[] => {
+  return glob.sync(
+    "{frontend,enterprise/frontend}/**/*.{css,module.css,js,jsx,ts,tsx}",
+  );
+};
+
+const extractVariableDefinitions = (filePath: string): Set<string> => {
+  const fileContent = readFileSync(filePath, "utf8");
+  return extractVariableDefinitionsFromFileContent(fileContent);
+};
+
+export const extractVariableDefinitionsFromFileContent = (
+  fileContent: string,
+): Set<string> => {
+  const patterns = [
+    /--[a-zA-Z0-9-]+:/g, // css files: --my-css-variable: blue
```

Comment:
would this one regex work instead of 2?

```regex
['"`]?--[\w-]+['"`]?:
```

I tried searching for `\s*` but it doesn't find any new results, so I didn't include it.

Reply:
I tested it and it found more variables, the issue is that they seem false positive, for example it found `--withSidebar` because it parsed it from `          "AutomaticDashboard--withSidebar": hasSidebar,`

Your version has optional quotes on both ends, but if a quote is present then it needs to be found on both ends

Reply:
>Your version has optional quotes on both ends, but if a quote is present then it needs to be found on both ends

Very good point to keep your 2 versions.